dotnet/src/Connectors/Connectors.Memory.AzureCosmosDBMongoDB/AzureCosmosDBMongoDBVectorStoreRecordCollection.cs

```diff
@@ -288,10 +292,27 @@ public async IAsyncEnumerable<VectorSearchResult<TRecord>> VectorizedSearchAsync
 
         var vectorPropertyName = this._storagePropertyNames[vectorProperty.DataModelPropertyName];
 
+        var filter = AzureCosmosDBMongoDBVectorStoreCollectionSearchMapping.BuildFilter(
+            searchOptions.Filter,
+            this._storagePropertyNames);
+
+        // Constructing a query to fetch "offset + limit" total items
+        // to perform offset logic locally, since offset parameter is not part of API. 
+        var itemsAmount = searchOptions.Offset + searchOptions.Limit;
```

Comment:
So this involves over fetching, is that right?

If Offset was 1000 and Limit was 2 we would fetch 1002 records and discard the first 1000. Is that correct?

Reply:
That's correct, because `Offset` is not available in Azure CosmosDB MongoDB vector search API as per documentation. `Filter` feature is in preview. So, we can ignore this parameter or throw an exception when it's specified to mention that it's not supported or simulate this logic locally.

Reply:
As per discussion, leaving over fetching logic for now.